### PR TITLE
rustdesk-server: init at 1.1.8

### LINKED ISSUE #243523: Package request: rustdesk-server
Self hosting server for Rust Desk remote desktop

the rustdesk client app is available, but it would be great to be able to self-host.

**Metadata**

* homepage URL: https://rustdesk.com/
* source URL: https://github.com/rustdesk/rustdesk-server
* license:  GNU AFFERO GENERAL PUBLIC LICENSE
* platforms: linux, windows, mac, android, ios

### PR DESCRIPTION
**Pull request by gaelreyrol**
###### Description of changes

This PR adds the package rustdesk-server: https://github.com/rustdesk/rustdesk-server

Closes https://github.com/NixOS/nixpkgs/issues/243523 https://github.com/NixOS/nixpkgs/issues/232445

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/rustdesk-server/default.nix [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  version = "1.1.8";
+
+  src = fetchFromGitHub {
+    owner = "rustdesk";
+    repo = "rustdesk-server";
+    rev = version;
+    hash = "sha256-KzIylbPe+YN513UNFuisvDAIe4kICGYDUrfURDOw/no=";
+  };
+
+  cargoLock = {
```
Comment (dasJ) [inline pkgs/servers/rustdesk-server/default.nix:23]: Why not specify the cargo hash rather than vendoring the file and adding the hashes for the git deps?

Comment (gaelreyrol) [reply pkgs/servers/rustdesk-server/default.nix:23]: It seems I cannot specify outputHashes without specifying a local lock file.

Comment (amz-x) [reply pkgs/servers/rustdesk-server/default.nix:23]: @dasJ I can confirm @gaelreyrol statement here, tried to build it by specifying the `cargoSha256` and the build fails with the following:

```
last 10 log lines:
       > building
       > source = "git+https://github.com/open-trade/async-speed-limit#f89f702ae01d4016429543d2f0dda1086157e420"
       > source = "git+https://github.com/open-trade/confy#630cc28a396cb7d01eefdd9f3824486fe4d8554b"
       > ERROR: The Cargo.lock contains git dependencies
       >
       > This is currently not supported in the fixed-output derivation fetcher.
       > Use cargoLock.lockFile / importCargoLock instead.
```

Looks like the **rustdesk** application is also using hashes for the git deps, see most recent [commit](https://github.com/NixOS/nixpkgs/commit/cf3c4da2ea84369ba2bdc8c412de01ddcb91d75c).

Comment (dasJ) [reply pkgs/servers/rustdesk-server/default.nix:23]: Damn.

**Comment by sjau** [comment]
There are no options for it e.g. port, data storage etc?

**Comment by devhell** [comment]
> There are no options for it e.g. port, data storage etc?

I was wondering the same thing. I would have thought this would be available as a service, like `services.jitsi-meet.*`, etc.

**Comment by gaelreyrol** [comment]
@sjau @devhell This PR adds the package, not the module. There is an on going PR on this: https://github.com/NixOS/nixpkgs/pull/272501.

**Pull request merged**